docs/k8s/subnet.yaml

```diff
@@ -1,31 +1,43 @@
 # Copyright (c) Tailscale Inc & AUTHORS
 # SPDX-License-Identifier: BSD-3-Clause
-apiVersion: v1
-kind: Pod
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
In the [Tailscale Kubernetes operator]() we create all standalone proxies as `Statefulset`s (this is mostly for future when we will be able to allow multiple replicas, but will likely need to ensure unique network identifiers and other features that come with `Statefulset`s.) I think we want to make this a `Statefulset` too - for consistency at least.